src/cargo/core/workspace.rs

```diff
@@ -222,6 +223,20 @@ impl<'cfg> Workspace<'cfg> {
         }
     }
 
+    /// Returns the root [augment] section of this workspace.
+    ///
+    /// This may be from a virtual crate or an actual crate.
+    pub fn root_augment(&self) -> &HashMap<Url, Vec<Dependency>> {
```

Comment:
Am I correct that `augment` is only allowed in the root of the workspace? We should probably enforce this.